src/transformers/models/bert/modeling_bert.py

```diff
@@ -192,10 +203,22 @@ def forward(
         seq_length = input_shape[1]
 
         if position_ids is None:
-            position_ids = self.position_ids[:, past_key_values_length : seq_length + past_key_values_length]
+            if hasattr(self, "position_ids"):
+                position_ids = self.position_ids[:, past_key_values_length : seq_length + past_key_values_length]
+            else:
+                position_ids = torch.arange(seq_length).expand((1, -1))
```

Comment:
I would just leave the position IDs as they are

Reply:
@LysandreJik Thanks for you reviewing the changes, do you mean in the else block ? as now we are registering the buffer for position_ids only if PT version is > 1.6 then we need to handle this case.  So should I use they way they are constructed in the constructor "torch.arange(config.max_position_embeddings).expand((1, -1))" ?